python/lsst/the/monster/utils.py

```diff
@@ -44,3 +46,75 @@ def read_stars(path, indices, allow_missing=False):
     stars["coord_dec"].convert_unit_to(units.degree)
 
     return stars
+
+
+def makeRefSchema(survey, bands, reference_name):
+    """
+    Make the refcat schema
+
+    Parameters
+    ----------
+    survey : `str`
+        Name of the survey whose refcat we are writing
+    bands : `List` of `str`
+        Names of the bands in the refcat
+    reference_name : `str`
+        Name of the overall reference catalog
+
+    Returns
+    -------
+    refSchema: `lsst.afw.table.Schema`
+    """
+
+    refSchema = afwTable.SimpleTable.makeMinimalSchema()
+    refSchema.addField(reference_name+'_id', type='L', doc='Gaia DR3 ID')
+
+    for band in bands:
+        colname = 'decam_'+band+'_from_'+survey+'_flux'
+        colname_err = colname+'Err'
+        refSchema.addField(colname, type='D',
+                           doc='flux transformed to DECam system')
+        refSchema.addField(colname_err, type='D',
+                           doc='error on flux transformed to DECam system')
+
+    return refSchema
+
+
+def makeRefCat(refSchema, refTable, survey, bands, reference_name):
+    """
+    Make the standard star catalog for persistence
+
+    Parameters
+    ----------
+    refSchema: `lsst.afw.table.Schema`
+       Standard star catalog schema
+    refTable: `Astropy Table`
+       Reference catalog to convert
+    survey : `str`
+        Name of the survey whose refcat we are writing
+    bands : `List` of `str`
+        Names of the bands in the refcat
+    reference_name : `str`
+        Name of the overall reference catalog
+
+    Returns
+    -------
+    refCat: `lsst.afw.table.BaseCatalog`
+       Standard star catalog for persistence
+    """
+
+    refCat = afwTable.SimpleCatalog(refSchema)
+    refCat.resize(np.size(refTable))
+
+    refCat['id'][:] = refTable['id']
+    refCat[reference_name+'_id'][:] = refTable[reference_name+'_id']
+    refCat['coord_ra'][:] = refTable['coord_ra'] * geom.degrees
```

Comment:
I'm surprised this works ... it's certainly going to be slow and inefficient because the geom units aren't vectorized at all.  This should be `refCat['coord_ra'][:] = np.deg2rad(refTable['coord_ra']`.  (And similar for the declination)